x-pack/plugins/fleet/server/services/epm/elasticsearch/template/install.ts

```diff
@@ -322,9 +322,9 @@ export function buildComponentTemplates(params: {
           ...templateSettings.index,
           ...(pipelineName ? { default_pipeline: pipelineName } : {}),
           mapping: {
-            ...templateSettings?.mapping,
+            ...templateSettings.index?.mapping,
```

Comment:
I guess this is the bug you mentioned? 

> I had to fix a bug that prevented index.mappings to be copied from the default settings into the template settings

What are other side effects of this bug? I'm suggesting to get this into a separate PR (even though it is just 1 line) so later it can be tracked separately in case it causes other issues.